src/router/validation/subscriptionValidator.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import Ajv from 'ajv';
+import { isEmpty, groupBy } from 'lodash';
+
+import { InvalidResourceError, Validator, Search } from 'fhir-works-on-aws-interface';
+import subscriptionSchema from './subscriptionSchema.json';
+
+export interface SubscriptionEndpoint {
+    endpoint: string | RegExp;
+    headers?: string[];
+    tenantId?: string;
+}
+
+const SUBSCRIPTION_RESOURCE_TYPE = 'Subscription';
+
+const BUNDLE_RESOURCE_TYPE = 'Bundle';
+
+const SINGLE_TENANT_ALLOW_LIST_KEY = 'SINGLE_TENANT_ALLOW_LIST_KEY';
+
+const isEndpointAllowListed = (allowList: (string | RegExp)[], endpoint: string): boolean => {
+    return allowList.some((allowedEndpoint) => {
+        if (allowedEndpoint instanceof RegExp) {
+            return allowedEndpoint.test(endpoint);
+        }
+        return allowedEndpoint === endpoint;
+    });
+};
+
+const extractSubscriptionResources = (resource: any): any[] => {
+    const { resourceType } = resource;
+    if (resourceType === SUBSCRIPTION_RESOURCE_TYPE) {
+        return [resource];
+    }
+    if (resourceType === BUNDLE_RESOURCE_TYPE) {
+        return resource.entry
+            .map((ent: { resource: any }) => ent.resource)
+            .filter(
+                (singleResource: { resourceType: string }) =>
+                    singleResource && singleResource.resourceType === SUBSCRIPTION_RESOURCE_TYPE,
+            );
+    }
+    return [];
+};
+
+export default class SubscriptionValidator implements Validator {
+    private ajv: Ajv.Ajv;
+
+    private readonly validateJSON: Ajv.ValidateFunction;
+
+    private search: Search;
+
+    private allowListMap: { [key: string]: (string | RegExp)[] } = {};
+
+    private readonly enableMultiTenancy: boolean;
+
+    constructor(search: Search, allowList: SubscriptionEndpoint[], enableMultiTenancy: boolean) {
```

Comment:
Can you make the last param an optional object and have `enableMultiTenancy` default to false?
e.g. https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/blob/e2ccf527eeb5fede5ac0a1b42511ec4ab08b7255/src/objectStorageService/s3DataService.ts#L42

The different classes that use `enableMultiTenancy` follow that pattern. Plus, it's easier to understand ( it's hard to tell the meaning of true in `new SubscriptionValidator(search, multiTenantAllowList, true);`)